src/main/java/org/jetbrains/kotlinx/lincheck/CTestStructure.java

```diff
@@ -172,7 +174,7 @@ private static void readTestStructureFromClass(Class<?> clazz, Map<String, Param
             if (name != null)
                 return checkAndGetNamedGenerator(namedGens, name);
             // Parameter generator is not specified, try to create a default one
-            ParameterGenerator<?> defaultGenerator = createDefaultGenerator(p);
+            ParameterGenerator<?> defaultGenerator = createDefaultGenerator(p, defaultGens);
```

Comment:
Let's create all default generators in advance and perform `defaultGens [p.getType()]!!` here